api/src/main/java/io/strimzi/api/kafka/model/listener/TlsListenerConfiguration.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.CertAndKeySecretSource;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Configures Tls listeners
+ */
+@JsonPropertyOrder({"serverKey"})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(
+    editableEnabled = false,
+    generateBuilderPackage = false,
+    builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@EqualsAndHashCode
+public class TlsListenerConfiguration implements Serializable, UnknownPropertyPreserving {
```

Comment:
I know that this class is meant to be used for the `tls` listener type while `KafkaExternalListenerConfiguration` for the `external` listener types ... but they are exactly the same :-)
I think we can factor out a common class with such a configuration for both different types.

Reply:
@stanlyDoge what about this suggested refactoring?